web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/arbeidInntektsmelding/ArbeidOgInntektsmeldingRestTjeneste.java

```diff
@@ -116,11 +118,19 @@ public List<InntektsmeldingDto> getAlleInntektsmeldinger(@TilpassetAbacAttributt
                                                                    @NotNull @QueryParam(UuidDto.NAME) @Parameter(description = UuidDto.DESC) @Valid UuidDto uuidDto) {
         var behandling = behandlingRepository.hentBehandling(uuidDto.getBehandlingUuid());
         var ref = BehandlingReferanse.fra(behandling);
-        var stp = skjæringstidspunktTjeneste.getSkjæringstidspunkter(behandling.getId());
+        var stp = Optional.ofNullable(safeSkjæringstidspunktOpptjening(behandling.getId())).map(Skjæringstidspunkt::getUtledetSkjæringstidspunkt).orElse(null);
 
         return arbeidOgInntektsmeldingDtoTjeneste.hentAlleInntektsmeldingerForFagsak(ref, stp);
     }
 
+    private Skjæringstidspunkt safeSkjæringstidspunktOpptjening(Long behandlingId) {
+        try {
+            return skjæringstidspunktTjeneste.getSkjæringstidspunkter(behandlingId);
```

Comment:
vurdert en optional  metode i tjenesten istf en try catch?

Reply:
@palfi tenker på et par alternativ
* En soft setting der et Skjæringstidspunkt kommer med mye nullverdier vil gi exceptions der man bruker verdien
* En annen getSTP-metode for Dtos mv som har mer soft-setting - her må konsumenter regne med at tidspunkt er null